payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonRunner.java

```diff
@@ -128,8 +129,10 @@ public PaymentStateContext buildPaymentStateContext(final boolean isApiPayment,
                                                         final Iterable<PluginProperty> properties,
                                                         final CallContext callContext,
                                                         final InternalCallContext internalCallContext) throws PaymentApiException {
+
         // Retrieve the payment id from the payment external key if needed
         final UUID effectivePaymentId = paymentId != null ? paymentId : retrievePaymentId(paymentExternalKey, paymentTransactionExternalKey, internalCallContext);
+        verifyPaymentMethodExistsForRefund(effectivePaymentId, transactionType, internalCallContext);
```

Comment:
Why do you need this here? Shouldn't it be done already when the payment system retrieves the plugin name (in `PaymentAutomatonDAOHelper` and/or in `ProcessorBase`)?